attorch/layers.py

```diff
@@ -405,28 +405,39 @@ def group_sparsity(self, group_size):
             ret = ret + (self.features[:, chunk:chunk + group_size, ...].pow(2).mean(1) + 1e-12).sqrt().mean() / n
         return ret
 
-    def forward(self, x, shift=None):
+    def forward(self, x, shift=None, out_idx=None):
         if self.positive:
             positive(self.features)
         self.grid.data = torch.clamp(self.grid.data, -1, 1)
         N, c, w, h = x.size()
         m = self.pool_steps + 1
         feat = self.features.view(1, m * c, self.outdims)
 
+        if out_idx is None:
+            grid = self.grid
+            bias = self.bias
+            outdims = self.outdims
+        else:
+            feat = feat[:, :, out_idx]
+            grid = self.grid[:, out_idx]
+            if self.bias is not None:
+                bias = self.bias[out_idx]
+            outdims = out_idx.size
```

Comment:
the propery `.size` would only work if `out_idx` is a NumPy array, and it would be awkward to restrict `out_idx`'s choice to that. Use `len(out_idx)` instead as `out_idx` should most definitely be any iterable list of integers and `len` should be defined anyway.